src/screens/FilterScreen.js

```diff
@@ -1,34 +1,198 @@
 import React, {Component} from 'react';
-import {Text} from 'react-native';
+import {Text, ScrollView, View, StyleSheet, Platform} from 'react-native';
+import {connect} from "react-redux";
 
+import _ from 'lodash';
 
+import {CheckBox, Button, Icon} from 'react-native-elements';
+import Accordion from 'react-native-collapsible/Accordion';
 
-import {CardSection, Card} from '../components/common/index';
-import MapScreen from "../components/SearchMap";
+import {checkboxWasClicked} from "../actions";
+
+import {navKeys} from '../constants';
+import {campsite, more_screen, reducerAlerts} from '../locale.en';
+import {campsiteIcon, grey} from "../styles";
+
+const {campsite_form: {accessibility, facilities, price, accessibility_options, facilities_options, price_options}} = campsite;
+
+
+const ACCESSIBILITY = [
+    {
+        title: accessibility,
+        content: accessibility_options
+    }
+];
+
+const FACILITIES = [
+    {
+        title: facilities,
+        content: facilities_options
+    }
+];
+
+const PRICE = [
+    {
+        title: price,
+        content: price_options
+    }
+];
 
 class FilterScreen extends Component {
 
+    componentWillMount() {
+        const {displaySites} = this.props;
+
+        this.props.navigation.setParams({siteCount: displaySites.length});
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const nextSetOfSites = nextProps.displaySites;
+        const currentSites = this.props.displaySites;
+
+
+        if (nextSetOfSites.length !== currentSites.length) {
+            this.props.navigation.setParams({siteCount: nextSetOfSites.length});
+        }
+    }
+
+    static renderRightNavButton = (navigate, {siteCount}) => {
+        if (Platform.OS === 'ios') {
+            return (
+                <Button
+                    title={_.isUndefined(siteCount) ? '' : `${siteCount} Results`}
+                    onPress={() => navigate(navKeys.SEARCH)}
+                    backgroundColor="rgba(0,0,0,0)"
+                    color="rgba(0,122,255,1)"
+                />
```

Comment:
Logic for the top right nav results button. If `siteCount` is `undefined` then show nothing. Otherwise we show the amount of results. I'll have to figure out how to localize the Results text where we can also pass a variable like this... but that's for another day...